maas/resource_maas_raid.go

```diff
@@ -0,0 +1,528 @@
+			AddBlockDevices:    newBlockDevice,
+			AddPartitions:      newPartition,
+			AddSpareDevices:    newSpareDevice,
+			AddSparePartitions: newSparePartition,
+		},
+	)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// 2. Remove any spare disks that are no longer in the RAID
+	//    (Including disks moving from spare to active, so that we do not have add/remove collisions)
+	if len(removedSpareDevice)+len(removedSparePartition) > 0 {
+		if _, err = client.RAID.Update(machine.SystemID, id,
+			&entity.RAIDUpdateParams{
+				RemoveSpareDevices:    removedSpareDevice,
+				RemoveSparePartitions: removedSparePartition,
+			},
+		); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	// 3. Re-add any disks that were moved from spare to active
+	if len(movedBlockDevice)+len(movedPartition) > 0 {
+		if _, err = client.RAID.Update(machine.SystemID, id,
+			&entity.RAIDUpdateParams{
+				AddBlockDevices: movedBlockDevice,
+				AddPartitions:   movedPartition,
+			},
+		); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	// 4. Remove any active disks that are no longer in the RAID
+	//    (Including disks moving from active to spare, so that we do not have add/remove collisions)
+	if len(removedBlockDevice)+len(removedPartition) > 0 {
+		if _, err = client.RAID.Update(machine.SystemID, id,
+			&entity.RAIDUpdateParams{
+				RemoveBlockDevices: removedBlockDevice,
+				RemovePartitions:   removedPartition,
+			},
+		); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	// 5. Re-add any disks that were moved from active to spare
+	if len(movedSpareDevice)+len(movedSparePartition) > 0 {
+		if _, err = client.RAID.Update(machine.SystemID, id,
+			&entity.RAIDUpdateParams{
+				AddSpareDevices:    movedSpareDevice,
+				AddSparePartitions: movedSparePartition,
+			},
+		); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	// We can finally perform mounting and formatting operations on the virtual block device
+	if _, err = formatAndMountVirtualBlockDevice(client, &raid.VirtualDevice, d); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%v", raid.ID))
+
+	return resourceRAIDRead(ctx, d, meta)
+}
+
+func resourceRAIDDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ClientConfig).Client
+
+	machine, err := getMachine(client, d.Get("machine").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = client.RAID.Delete(machine.SystemID, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func verifyRAIDConfig(client *client.Client, machine *entity.Machine, d *schema.ResourceData) error {
+	// Ensure the provided config has the correct disks for the raid level, that each block device is partitionless,
+	// that the boot-disk is not provided while block devices are (see: VolumeGroups for similar behaviour), and that
```

Comment:
US English is preferred: 
```suggestion
	// that the boot-disk is not provided while block devices are (see: VolumeGroups for similar behavior), and that
```